docs/federation.md

```diff
@@ -0,0 +1,116 @@
+Administrator”).
+
+## Principles of Federation
+
+The following sections describe the requirements for operating and maintaining a Node.
+
+### Operations
+
+- Deployment of software stack: Each Node should deploy the DACCS software stack. Node Administrators are required to
+  upgrade the software stack to the latest DACCS stack within two weeks of the updates release date.
+- Node maintenance: Node Administrators accept that joining the Network implies that they are responsible for
+  provisioning, funding and maintaining their own Node.
+- Node security: Node Administrators are responsible for maintaining secure access to their node and for coordinating
+  with
+  the hosting institution regarding any security related considerations prior to the deployment of the Node. Admin
+  access
+  to the Node must be restricted to listed Node Administrators.
+- Downtime policy: A Node should strive to maintain downtime of no more than 5%. If a Node needs to be taken offline for
+  scheduled maintenance purposes, the Node Administrator must provide users at least 2-weeks notice.
+- New user signups: Node Administrators are responsible for the signups of users on their own Nodes. All new signup
+  requests should be verified and approved or rejected within 2 business days.
+- Node configuration: The DACCS deployment stack consists of core components and optional components. A Node can enable
+  or
+  disable non-core components as per the decisions of the Node Administrator. However, when a feature/component is to be
+  removed, adequate notice for the change should be given to the Node’s users.
+
+### Support and Documentation
+
+A Node should maintain a public web-page with:
+
+1. basic information about the Node,
+2. instructions for new users to
+   sign up for an account at the Node,
+3. basic usage policies (as determined by the Node Administrator), and
+4. support information.
+
+We encourage Node Administrators to develop additional documentations and tutorials, if they can,
+as it will be of great value to users on that Node and to other users on the Network, but we do not require it.
+
+Node Administrators are expected to provide technical support to users of their Nodes in a timely manner.
+
+### User Data and Privacy
+
+- Privacy: No Personal Identifying Information belonging to a User can be made accessible to anyone except the User in
+  question, the Node Manager and the DACCS Executive Committee (“Executive Committee”) without explicit written consent
+  of
+  the User in question.
+- No User Generated Data can be made accessible to another User without the User’s permission. without the user’s
+  permission. User Generated Data can not be made accessible outside the Network without the User’s permission.
+- Back-up of User Generated Data: The Node Administrator should have a mechanism in place to regularly back up a user’s
+  main
+  working directory, and have a mechanism in place to retrieve the back-up data upon a user’s request.
```

Comment:
Ouranos currently do not provide any data backup to our Jupyter users.  Data retention is basically as long as they are active Jupyter users.  Otherwise we can also delete their account to reclaim disk space.

Users are responsible to export their code or data output if they need to preserve them.

We also ask our users to be sensible to the amount of disk space, CPU, memory they use.  The platform is not an HPC.

Can each node have their own backup and data retention policy or all should follow the same policy?

Reply:
Each node can have its own policy. We're trying to outline minimum requirements here.

> Data retention is basically as long as they are active Jupyter users.

Ok, but what happens if you're storing active jupyter workspaces on a disk and that disk fails? Do you have any backups or RAIDing of those disks?

Reply:
No backup but our disks are RAID.

Reply:
I'm happy to say:

- minimal requirement: RAID (or some other protection from disk failure)
- recommended requirement: full backups with a defined recovery policy

what do you think?

Reply:
Good for me.